02_knight_and_princess/tevfik_aksoy/main.cpp

```diff
@@ -121,13 +145,23 @@ std::string Edge::to_JSON() const {
   return json_string;
 }
 
+bool validation::does_vertex_exist(const VertexId& id,
+                                   const vector<Vertex>& vertices) {
+  for (const auto& vertex : vertices)
+    if (vertex.id == id) {
+      return true;
+    }
+  return false;
+}
+
 class Graph {
  public:
   VertexId insert_vertex() {
     const auto id = get_new_vertex_id_();
     vertices_.emplace_back(id);
     if (id == 0) {
-      depth_map_[0].push_back(0);
+      depth_map_.resize(1);
+      depth_map_[0].push_back(id);

```

Comment:
уж лучше напиши
```suggestion
      auto& first_depth_layer = depth_map_.emplace_back({});
      first_depth_layer.push_back(id);
```
Так будет более наглядно, что мы добавляем новый пустой вектор

Reply:
`depth_map_.emplace_back({})` выдает ошибку -> `depth_map_.emplace_back();`